### PR TITLE
Authentication/Profile Fixes

### PR DESCRIPTION
**Pull request by zeitschlag**
- We missed a `MastodonUser.mastodonAuthentication`. The consequence of this was that I couldn't edit my profile anymore
- I also fixed a warning (and a crash when I tried to reorder the fields)

### CODE REVIEW

File: MastodonSDK/Sources/CoreDataStack/CoreData.xcdatamodeld/CoreData 9.xcdatamodel/contents [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
-<model type="com.apple.IDECoreDataModeler.DataModel" documentVersion="1.0" lastSavedToolsVersion="22222" systemVersion="23B74" minimumToolsVersion="Automatic" sourceLanguage="Swift" userDefinedModelVersionIdentifier="">
+<model type="com.apple.IDECoreDataModeler.DataModel" documentVersion="1.0" lastSavedToolsVersion="22222" systemVersion="22G74" minimumToolsVersion="Automatic" sourceLanguage="Swift" userDefinedModelVersionIdentifier="">
```
Comment (DK1DA) [inline MastodonSDK/Sources/CoreDataStack/CoreData.xcdatamodeld/CoreData 9.xcdatamodel/contents]: @zeitschlag hm, well this is annoying as it apparently stored your macOS build version here ...

File: Mastodon/Scene/Profile/ProfileViewModel.swift [hunk 2]
```diff
@@ -178,12 +178,11 @@ class ProfileViewModel: NSObject {
 
     // fetch profile info before edit
     func fetchEditProfileInfo() -> AnyPublisher<Mastodon.Response.Content<Mastodon.Entity.Account>, Error> {
-        guard let me = me,
-              let mastodonAuthentication = me.mastodonAuthentication
-        else {
+        guard let me else {
```
Comment (DK1DA) [inline Mastodon/Scene/Profile/ProfileViewModel.swift:181]: ❤️ 💯

**Review by kimar — APPROVED** [review]
Works like a charm now 👍

File: MastodonSDK/Sources/CoreDataStack/CoreData.xcdatamodeld/CoreData 9.xcdatamodel/contents (hunk 1 above)
Comment (zeitschlag) [reply MastodonSDK/Sources/CoreDataStack/CoreData.xcdatamodeld/CoreData 9.xcdatamodel/contents]: Core Data is doing Core Data things 😆

I should upgrade to a most recent version 🤔

**Pull request merged**